cmds/rush/readline.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/chzyer/readline"
+)
+
+func usage(w io.Writer) {
+	io.WriteString(w, "commands:\n")
+	io.WriteString(w, completer.Tree("    "))
+}
+
+// Function constructor - constructs new function for listing given directory
+func listFiles(path string) func(string) []string {
+	return func(line string) []string {
```

Comment:
which makes sense because the files are not static, right? You want to be able to call this function over and over in the event the files change. It's actually quite nice the way it works.